src/backend/distributed/utils/maintenanced.c

```diff
@@ -246,6 +246,17 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 	/* make worker recognizable in pg_stat_activity */
 	pgstat_report_appname("Citus Maintenance Daemon");
 
+	/* before doing the actual jobs, check any version mismatches */
+	StartTransactionCommand();
+
+	if (!CheckCitusVersion(LOG))
+	{
+		/* return code of 1 requests worker restart */
+		proc_exit(1);
```

Comment:
> This'll spam the log a bit

Hmm, that makes sense. But still I'd prefer to do the check before the loop starts. I'll think more and try to come up with a better solution.